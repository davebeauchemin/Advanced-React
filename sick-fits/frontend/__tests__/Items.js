import ItemComponent from '../components/Item';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'ABCD123',
  title: 'Cool Item',
  price: 5000,
  description: 'This item is really cool',
  image: 'dog.jpg',
  largeImage: 'largedog.jpg',
}

describe('<Item/>', () => {
  it('renders and match the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});