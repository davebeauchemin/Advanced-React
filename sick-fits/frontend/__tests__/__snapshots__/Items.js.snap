// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and match the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABCD123",
          },
        }
      }
    >
      <a>
        Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is really cool
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABCD123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABCD123"
    />
    <DeleteItem
      id="ABCD123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
